sqlglot/dataframe/sql/column.py

```diff
@@ -0,0 +1,330 @@
+
+    def __invert__(self):
+        return self.unary_op(exp.Not)
+
+    def __rand__(self, other: ColumnOrPrimitive) -> Column:
+        return self.inverse_binary_op(exp.And, other)
+
+    def __ror__(self, other: ColumnOrPrimitive) -> Column:
+        return self.inverse_binary_op(exp.Or, other)
+
+    @classmethod
+    def ensure_col(cls, value: t.Union[ColumnOrName, int, float]):
+        return cls(value)
+
+    @classmethod
+    def ensure_cols(cls, args: t.List[t.Union[int, str, Column]]) -> t.List[Column]:
+        return [cls.ensure_col(x) if not isinstance(x, Column) else x for x in args]
+
+    @classmethod
+    def _lit(cls, value: t.Optional[t.Any] = None) -> Column:
+        if value is None:
+            return cls(exp.Null())
+        if isinstance(value, (str, int, bool, float)):
+            value_formatted = str(value).lower() if isinstance(value, bool) else str(value)
+            return cls(exp.Literal(this=value_formatted, is_string=isinstance(value, str)))
+        if isinstance(value, dict):
+            columns = [cls._lit(v).alias(k).expression for k, v in value.items()]
+            return cls(exp.Struct(expressions=columns))
+        if isinstance(value, Iterable):
+            expressions = [cls._lit(x).expression for x in value]
+            return cls(exp.Column(this=exp.Array(expressions=expressions)))
+        if isinstance(value, datetime.datetime):
+            datetime_literal = exp.Literal(this=value.strftime("%Y-%m-%d %H:%M:%S"), is_string=True)
+            return cls(
+                exp.Column(
+                    this=exp.StrToTime(
+                        this=datetime_literal, format=exp.Literal(this="YYYY-MM-DD HH:MM:SS", is_string=True)
+                    )
+                )
+            )
+        if isinstance(value, datetime.date):
+            date_literal = exp.Literal(this=value.strftime("%Y-%m-%d"), is_string=True)
+            return cls(
+                exp.Column(this=exp.StrToDate(this=date_literal, format=exp.Literal(this="YYYY-MM-DD", is_string=True)))
+            )
+        return cls(value)
+
+    @classmethod
+    def invoke_anonymous_function(cls, column: t.Union[ColumnOrName, None], func_name: str, *args) -> Column:
+        column = [cls.ensure_col(column)] if column is not None else []
+        args = [cls.ensure_col(arg) for arg in args]
+        expressions = [x.expression for x in column + args]
+        new_expression = exp.Anonymous(this=func_name.upper(), expressions=expressions)
+        return Column(new_expression)
+
+    @classmethod
+    def invoke_expression_over_column(
+        cls, column: t.Union[ColumnOrName, None], callable_expression: t.Callable, **kwargs
+    ) -> Column:
+        column = cls.ensure_col(column) if column is not None else None
+        new_expression = (
+            callable_expression(this=column.column_expression, **kwargs)
+            if column is not None
```

Comment:
for all of these i prefer the inverse which is

if column is None else ... you have a couple of these calls